src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs

```diff
@@ -356,6 +359,10 @@ private void HandleReplicationErrors(Action replicationAction)
                 _parent.ForTestingPurposes?.OnOutgoingReplicationStart?.Invoke(this);
                 replicationAction();
             }
+            catch (MissingAttachmentException e)
+            {
+                HandleException(e);
```

Comment:
don't we need to check here `MissingAttachmentsRetries > 1` ? we want to start delaying if we actually have a loop, no?

Reply:
The only case we end up here is if `MissingAttachmentsRetries > 1` and we throw this exception from `HandleServerResponse(BlittableJsonReaderObject replicationBatchReplyMessage, bool allowNotify)`